tools/istio-iptables/pkg/dependencies/implementation_linux.go

```diff
@@ -82,72 +61,73 @@ func (r *RealDependencies) execute(cmd string, ignoreErrors bool, stdin io.Reade
-		externalCommand := exec.Command(cmd, args...)
-		stdout = &bytes.Buffer{}
-		stderr = &bytes.Buffer{}
-		externalCommand.Stdout = stdout
-		externalCommand.Stderr = stderr
-		externalCommand.Stdin = stdin
-		if stdin != nil {
-			if _, err := stdin.Seek(0, io.SeekStart); err != nil {
-				return err
+	return c.Run()
+}
+
+var (
+	// IptablesRestoreLocking is the version where locking and -w is added to iptables-restore
+	IptablesRestoreLocking = utilversion.MustParseGeneric("1.6.2")
+	// IptablesLockfileEnv is the version where XTABLES_LOCKFILE is added to iptables.
+	IptablesLockfileEnv = utilversion.MustParseGeneric("1.8.6")
+)
+
+func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, stdin io.ReadSeeker, args ...string) error {
+	mode := "without lock"
+	var c *exec.Cmd
+	_, needLock := XTablesWriteCmds[cmd]
+	if !needLock || r.IptablesVersion.NoLocks() {
+		// No locking supported/needed, just run as is. Nothing special
+		c = exec.Command(cmd, args...)
+	} else {
+		if r.CNIMode {
+			// In CNI, we are running the pod network namespace, but the host filesystem. Locking the host is both useless and harmful,
+			// as it opens the risk of lock contention with other node actors (such as kube-proxy), and isn't actually needed at all.
+			// In both cases we are setting the lockfile to `r.NetworkNamespace`.
+			// * /dev/null looks like a good option, but actually doesn't work (it will ensure only one actor can access it)
+			// * `mktemp` works, but it is slightly harder to deal with cleanup and in some platforms we may not have write access.
+			if r.IptablesVersion.version.LessThan(IptablesLockfileEnv) {
+				// Older iptables cannot turn off the lock explicitly, so we hack around it...
+				// Overwrite the lock file with /dev/null
+				// cmd is repeated twice as the first 'cmd' instance becomes $0
+				args := append([]string{"-c", fmt.Sprintf("mount --bind %s /run/xtables.lock; exec $@", r.NetworkNamespace), cmd, cmd}, args...)
+				c = exec.Command("sh", args...)
+				// Run in a new mount namespace so our mount doesn't impact any other processes.
+				c.SysProcAttr = &syscall.SysProcAttr{Unshareflags: unix.CLONE_NEWNS}
+				mode = "without lock by mount"
+			} else {
+				// Available since iptables 1.8.6+, just point to a different file directly
+				c = exec.Command(cmd, args...)
+				c.Env = append(c.Env, "XTABLES_LOCKFILE="+r.NetworkNamespace)
+				mode = "without lock by environment"
 			}
-		}
-		if r.CNIMode && !r.HostNSEnterExec {
-			err = nsContainer.Do(func(ns.NetNS) error {
-				return externalCommand.Run()
-			})
 		} else {
-			err = externalCommand.Run()
-		}
-		exitCode, ok := exitCode(err)
-		if !ok {
-			// cannot get exit code. consider this as non-retriable.
-			return nil
-		}
-
-		if !isXTablesLockError(stderr, exitCode) {
-			// Command succeeded, or failed not because of xtables lock.
-			return nil
+			// We want the lock. Wait up to 30s for it.
+			args = append(args, "--wait=30")
```

Comment:
30 seconds is a lot ... kubernetes considers only 5 seconds for the wait 

https://github.com/kubernetes/kubernetes/blob/ba0505319e11f9b9cae458b41b8adfb73b97a312/pkg/util/iptables/iptables.go#L189